osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,125 @@
+    # Retrieval of tasks that has not been applied to retention policy to be applied
+    tasks = Task.objects.filter(state__in=FINISHED_STATES).exclude(id__in=processed_tasks)
+
+    # ---------- Section to take parameters from DB
+    retention_personal_task_parameter_exists = False
+    try:
+        retention_personal_tasks = RetentionPolicySetting.objects.get(category="PersonalScan")
+        retention_personal_task_parameter_exists = True
+    except RetentionPolicySetting.DoesNotExist:
+        logger.warning("Retention policy setting(PersonalScan) object does not exist. These tasks will not be deleted.")
+    except DatabaseError as e:
+        logger.error("Error obtaining PersonalScan value %s.", e)
+        sys.exit(1)
+
+    failed_status_task_parameter_exists = False
+    try:
+        retention_failed_status = RetentionPolicySetting.objects.get(category="FailedStatus")
+        failed_status_task_parameter_exists = True
+    except RetentionPolicySetting.DoesNotExist:
+        logger.warning("Retention policy setting object(FailedStatus) does not exist. These tasks will not be deleted.")
+    except DatabaseError as e:
+        logger.error("Error obtaining FailedStatus value: %s", e)
+        sys.exit(1)
+    # ---------- End Section to take parameters from DB
+
+    eligible_tasks_to_be_deleted = []
+
+    # We identify which tasks are eligible to be deleted
+    today = date.today()
+    for t in tasks:
+        # At the moment, we only have 2 rules for retention settings:
+        # Personal tasks
+        # Failed status
+        # If dt_finished is None, the task is being executed at the moment
+        if t.dt_finished is None:
+            continue
+
+        if retention_personal_task_parameter_exists:
+            calculated_date_personal_tasks = (t.dt_finished + datetime.timedelta(days=retention_personal_tasks.days))\
+                .date()
+
+        if failed_status_task_parameter_exists:
+            calculated_date_status_failed = (t.dt_finished + datetime.timedelta(days=retention_failed_status.days))\
+                .date()
+
+        if retention_personal_task_parameter_exists and identify_personal_task(t.owner.username) and \
+                calculated_date_personal_tasks <= today:
+            # Personal Scans eligible to be deleted
+            eligible_tasks_to_be_deleted.append([t, retention_personal_tasks])
+            logger.debug('The following "Personal Scan" task results will be deleted: %s', t.id)
+        elif failed_status_task_parameter_exists and t.is_failed() and calculated_date_status_failed <= today:
+            # Failed Scans eligible to be deleted
+            eligible_tasks_to_be_deleted.append([t, retention_failed_status])
+            logger.debug('The following "Failed Scan" task results will be deleted: %s', t.id)
+
+    delete_tasks_results(eligible_tasks_to_be_deleted)
+    logger.info("Finishing Retention Policy Enforcement")
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
In this file, some lines are too long, I think it is better less than 100

Reply:
Fixed!